File: classical/CODE1.py
class Complex:
    def __init__(self, real, imag):
        self.real = real
        self.imag = imag

    def __str__(self) -> str:
        return f'{self.real} + i{self.imag}'


def div_complex(left, right) -> Complex:
    divisor = right.real**2 + right.imag**2
    a_top = left.real * right.real + left.imag * right.imag
    b_top = left.imag * right.real + left.real * right.imag

    return Complex(a_top/divisor, b_top/divisor)


if __name__ == '__main__':
    pass

File: classical/CODE1.cpp
#include <iostream>
#include <array>

std::array<int, 2> divComplex(std::array<int, 2> left, std::array<int, 2> right) {
    int divisor = right[0]*right[0] + right[1]*right[1];
    int a_top = left[0]*right[0] + left[1]*right[1];
    int b_top = left[1]*right[0] + left[0]*right[1];

    return std::array<int, 2>{a_top/divisor, b_top/divisor};
}


int main() {
    return 0;
}